### ISSUE TITLE
[abrt] shutter: g_type_check_instance_cast(): perl killed by SIGSEGV

### ISSUE THREAD

**Issue by martinkg** [shutter-project/shutter#535]
#### Brief summary of issue
Description of problem:
I was busy using Shutter. When I cancelled one of my actions, it crashed. I
believe 'window under cursor' was the action I cancelled.

#### Version-Release number of selected component:
shutter-0.99.2-4.fc37

#### Error output
reporter:       libreport-2.17.4
backtrace_rating: 4
cgroup:         0::/user.slice/user-1000.slice/session-2.scope
cmdline:        /usr/bin/perl /usr/bin/shutter --min_at_startup
crash_function: g_type_check_instance_cast
executable:     /usr/bin/perl
journald_cursor:
s=e57a4b5a627d4f2abffe13a0dee868d0;i=f0770;b=5a7d9971aaba4bc6af820259df20ed7c;m=3a15ebe12a;t=5f194fff69be0;x=8caf0fac4f1a17f8
kernel:         6.0.15-300.fc37.x86_64
rootdir:        /
runlevel:       N 5
type:           CCpp
uid:            1000

Truncated backtrace:
Thread no. 1 (10 frames)
 #0 g_type_check_instance_cast at ../gobject/gtype.c:4127
 #1 window_weak_notify_cb at ../libwnck/class-group.c:493
 #2 weak_refs_notify at ../gobject/gobject.c:3268
 #3 g_data_set_internal at ../glib/gdataset.c:410
 #4 g_object_real_dispose at ../gobject/gobject.c:1364
 #7 XS_Glib__Object_DESTROY at
/usr/src/debug/perl-Glib-1.3293-9.fc37.x86_64/GObject.xs:1301
 #8 Perl_pp_entersub at
/usr/src/debug/perl-5.36.0-492.fc37.x86_64/pp_hot.c:5352
 #9 Perl_call_sv at /usr/src/debug/perl-5.36.0-492.fc37.x86_64/perl.c:3057
 #10 S_curse at /usr/src/debug/perl-5.36.0-492.fc37.x86_64/sv.c:6983
 #11 Perl_sv_clear at /usr/src/debug/perl-5.36.0-492.fc37.x86_64/sv.c:6546

**Comment by Photon89** [comment]
Did it happen only once, multiple times or is it a reproducible crash? Did you use the Esc button to cancel the action?

**Comment by martinkg** [comment]
the user has answered that:
> I just tried to reproduce to the best of my knowledge. So far, I wasn't able to. I cannot remember at what stage I cancelled and if used ESC or Cancel button for that.

**Comment by Photon89** [comment]
I see. Please add whatever information might be added. Also, Shutter's own output would probably be more helpful than the backtrace, as the backtrace doesn't seem to include any of Shutter's functions.

**Comment by penguinpee** [comment]
I reported the issue in Fedora's Bugzilla using ABRT. I have since been unable to reproduce. Maybe this is a red herring and Shutter was crashing due to something else failing.

Where would i find Shutter's own output? I can take a look if it contains anything related to the issue.

**Comment by Photon89** [comment]
You would need to start Shutter from command line but this isn't practical if the issue is hard to catch.

You could write a script which starts Shutter and redirects the output into a log file, something like

```
#!/bin/bash

shutter > ~/shutter-log.log 2>&1
```

and create a starter which calls the script.

**Comment by Photon89** [comment]
@penguinpee Could you catch the issue again?

**Comment by penguinpee** [comment]
So far, the issue hasn't occurred again. But I'm having issues now with shutter not being started after login. The command used is `shutter --min_at_startup`. Clearly that's a different issue. I can open another issue for that.

I will modify the command used, so the output is saved to a log file. That should give me more insight. I will let you know if it happens again and hopefully have some output when it does.
